### DISCUSSION TITLE
How can I set language via URL

### DISCUSSION THREAD

**Discussion by leoavelino7** [sergiodxa/remix-i18next#85] [Q&A]
I created a base project using Remix + remix-18next where I set the language without flagging in the URL. [Project](https://github.com/leoavelino7/remix-i18n)

I would like the language to be set via the URL, how can I do that?

Example:
 - http://localhost:3000/pt-br/
 - http://localhost:3000/en/

**Comment by sergiodxa** [comment]
Create a folder like `routes/$lng` and inside place all your routes, now you can do `params.lng` to access the user locale.

So instead of using the getLocale function to know the user locale on the root loader, you will have to check the route params for that.

**Reply by JulienHe** [reply]
How can you set a default language without duplicating files?
Let's say I want to have

http://localhost:3000/ => Default to French
http://localhost:3000/pt-br/ => Brazilian Portuguese
http://localhost:3000/en/ => English
http://localhost:3000/fr/ => French or even redirect to /

How do you think we can achieve this 🤔 ??

**Reply by sergiodxa** [reply]
I would add a routes/$.tsx route and redirect to the default language
```ts
export async functions loader({ request }: LoaderArgs) {
  let url = new URL(request.url)
  if (pathname.startsWith(/fr|pt-br|en/)) throw json({}, 404)
  url.pathname = `/fr/${pathname}`
  return redirect(url.toString())
}
```
The $.tsx route will create a splat or catch-all route, every route not caught by another route file will go there.
